### PR TITLE
BUG: Fixed NDFrame.transform('abs')

### PR DESCRIPTION
**Pull request by TomAugspurger**
Closes https://github.com/pandas-dev/pandas/issues/19760

So this is a hacky workaround. I think we would ideally whitelist a set of valid NDFrame.transform methods, but I don't want to break things. What would that list include?

- abs
- pct_change
- shift
- tshift
- cum*

### CODE REVIEW

**Comment by WillAyd** [comment]
I'd also add fill and rank to your list. FWIW there's probably a lot of overlap with the groupby module and what it considers to be a transformation vs an aggregation. Wonder if we should keep all of these functions and their "application type" in a shared module for both DataFrame and GroupBy ops

File: pandas/core/apply.py [hunk 1]
```diff
@@ -111,7 +111,9 @@ def get_result(self):
 
         # string dispatch
         if isinstance(self.f, compat.string_types):
-            self.kwds['axis'] = self.axis
+            if self.f not in {'abs'}:
```
Comment (jreback) [inline pandas/core/apply.py]: NO, anytime 'hacky' work-around you need to re-think the patch.

Comment (jreback) [reply pandas/core/apply.py]: either you need signature introspection (in python 3, may not be possible in python 2), alternatively can catch this type of error, but should be much higher level (in .aggregate), and not here.

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/core/apply.py (hunk 1 above)
Comment (TomAugspurger) [reply pandas/core/apply.py]: Ugh, signature introspection seems hackier than hard-coding, no?

**Comment by TomAugspurger** [comment]
This should be for 0.23. It's a regression and was included in the docs.

File: pandas/core/apply.py (hunk 1 above)
Comment (TomAugspurger) [reply pandas/core/apply.py]: I'll see how hard it is to get Py2 & 3 working by inspecting the signature.

File: pandas/core/apply.py [hunk 2]
```diff
@@ -111,8 +111,11 @@ def get_result(self):
 
         # string dispatch
         if isinstance(self.f, compat.string_types):
-            self.kwds['axis'] = self.axis
-            return getattr(self.obj, self.f)(*self.args, **self.kwds)
+            func = getattr(self.obj, self.f)
```
Comment (jreback) [inline pandas/core/apply.py]: much better, and now IIRC we actually do something like this in groupby apply (for the same reason). can you add some comments here on what is going on

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/tests/frame/test_apply.py [hunk 3]
```diff
@@ -880,6 +880,13 @@ def f():
             with np.errstate(all='ignore'):
                 df.agg({'A': ['abs', 'sum'], 'B': ['mean', 'max']})
 
+    def test_transform_abs_name(self):
```
Comment (jreback) [inline pandas/tests/frame/test_apply.py]: do we need more tests? (for other funcs)

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/tests/frame/test_apply.py [hunk 4]
```diff
@@ -880,11 +881,14 @@ def f():
...
+    @pytest.mark.parametrize('method', [
+        'abs', 'shift', 'pct_change', 'cumsum', 'rank',
+    ])
+    def test_transform_method_name(self, method):
         # https://github.com/pandas-dev/pandas/issues/19760
         df = pd.DataFrame({"A": [-1, 2]})
-        result = df.transform('abs')
-        expected = pd.DataFrame({"A": [1, 2]})
+        result = df.transform(method)
+        expected = operator.methodcaller(method)(df)
```
Comment (WillAyd) [inline pandas/tests/frame/test_apply.py]: Learned something new today. Just out of curiosity, is this any different from doing `getattr(df, method)()`?

File: pandas/tests/frame/test_apply.py [hunk 5]
```diff
@@ -880,11 +881,14 @@ def f():
             with np.errstate(all='ignore'):
                 df.agg({'A': ['abs', 'sum'], 'B': ['mean', 'max']})
 
-    def test_transform_abs_name(self):
+    @pytest.mark.parametrize('method', [
+        'abs', 'shift', 'pct_change', 'cumsum', 'rank',
```
Comment (WillAyd) [inline pandas/tests/frame/test_apply.py]: Could also add `ffill` and `bfill`

File: pandas/tests/frame/test_apply.py (hunk 4 above)
Comment (TomAugspurger) [reply pandas/tests/frame/test_apply.py]: in this case, no. methodcaller is nicer though when you need to bind parameters, but don't have the object yet (common in pytest fixtures).

**Pull request merged**